source/grid/grid_generator.cc

```diff
@@ -2861,7 +2861,19 @@ namespace GridGenerator
       SubCellData());       // no boundary information
   }
 
-
+  template <int dim, int spacedim>
+  void
+  hyper_ball (Triangulation<dim,spacedim> &tria,
+              const Point<spacedim>   &p,
+              const double      radius)
+  {
+    Triangulation<spacedim> volume_mesh;
+    GridGenerator::hyper_ball(volume_mesh,p,radius);
+    std::set<types::boundary_id> boundary_ids;
+    boundary_ids.insert (0);
+    GridGenerator::extract_boundary_mesh (volume_mesh, tria,
+                                          boundary_ids);
```

Comment:
uh oh, I now understand. When you say `hyperball<2,3>` you mean the *surface* of the ball, right? I think then you're using the wrong word. In mathematical English, a *ball* are all the points `||x|| <= r` whereas a *sphere* are all the points with `||x||=r`. In other words, what you're creating here is a hypersphere. I think that's how you should name the function.